libs/product/src/selectors/composite-product/composite-product.selectors.ts

```diff
@@ -236,35 +230,64 @@ const createCompositeProductSelectors = (): DaffCompositeProductMemoizedSelector
 				return undefined;
 			}
 			const appliedOptions = selectCompositeProductAppliedOptions.projector(products, appliedOptionsEntities, { id: props.id });
-			const productDiscountAmount = selectProductDiscountAmount.projector(products, { id: props.id });
-			return daffAdd(productDiscountAmount, getOptionsDiscountAmount(<DaffCompositeProduct>product, appliedOptions));
+			return (<DaffCompositeProduct>product).items.reduce((acc, item) => daffAdd(
+				acc, 
+				appliedOptions[item.id] ? 
+					daffMultiply(
+						daffSubtract(appliedOptions[item.id].price, appliedOptions[item.id].discount && appliedOptions[item.id].discount.amount), 
+						appliedOptions[item.id].quantity
+					) : 
+					getMinimumRequiredCompositeItemDiscountedPrice(item)
+			), daffSubtract(product.price, product.discount ? product.discount.amount : 0));
```

Comment:
Same here, I think its better to null check outside of `daffSubtract` and default to `product.price`.

Reply:
done